#============================================================
# Patch to fix issue with KqueueSelector on macOS
# Upstream patch is pending
#
# Issue:  https://github.com/firecat53/urlscan/issues/155
#============================================================
diff --git a/urlscan/urlchoose.py b/urlscan/urlchoose.py
index 1dd3fd1..0783fa3 100644
--- urlscan/urlchoose.py.orig
+++ urlscan/urlchoose.py
@@ -36,6 +36,8 @@ import urwid
 if platform == 'darwin':
     COPY_COMMANDS = ('pbcopy',)
     COPY_COMMANDS_PRIMARY = ('pbcopy',)
+    import selectors
+    selectors.DefaultSelector = selectors.SelectSelector
 elif 'WAYLAND_DISPLAY' in os.environ:
     COPY_COMMANDS = ('wl-copy',)
     COPY_COMMANDS_PRIMARY = ('wl-copy --primary',)
